### PR TITLE
Fix #6: Use String instead of UTF8String

### LINKED ISSUE #6: Warning in Julia v0.6
Could you please fix this warning and tag a new version?

```julia
julia> topology_load()
WARNING: Compat.UTF8String is deprecated, use String instead.
```

### PR DESCRIPTION
**Pull request by juliohm**
Could you please review, merge and tag?

### CODE REVIEW

**Comment by juliohm** [comment]
Could someone please merge this? I don't think the error in AppVeyor is related to the changes I made.

File: src/Hwloc.jl [hunk 1]
```diff
@@ -207,7 +207,7 @@ end
 type Object
     type_::Symbol
     os_index::Int
-    name::Compat.UTF8String
+    name::Compat.String
```
Comment (eschnett) [inline src/Hwloc.jl:210]: Why is this change necessary?
Will the code still work with Julia 0.5 after this change?
Are you interested in becoming a maintainer of Hwloc.jl?

Comment (juliohm) [reply src/Hwloc.jl:210]: Hi @eschnett, I made the change based on the warning from Compat that suggested replacing UTF8String with String. The Travis CI build shows that the change works in Julia v0.5. By maintainer you mean someone that can merge PRs? I would be interested, but I will definitely need someone else's approval if I make any major change in the code.

**Comment by juliohm** [comment]
Thank for merging the changes. :blush: Could you please tag a new version? Some users of my package are complaining about the warnings.

**Comment by eschnett** [comment]
I'm not having much luck with tagging new versions recently. Usually, something doesn't work, and I have to perform manual git surgery. I'm quite annoyed by this, and can't wait for `Pkg3`.

Instead, I'll give you write access to the repository, allowing you to tag and publish a new minor version.

**Comment by juliohm** [comment]
Thank you @eschnett , I will configure attobot if it is not configured already to facilitate the tagging process.

**Comment by tkelman** [comment]
You probably don't want to make this change on julia 0.4, as it makes a field type non concrete

**Comment by juliohm** [comment]
@tkelman could you please elaborate? What would be a better solution?

**Comment by tkelman** [comment]
On the last 0.4-supporting version of Compat.jl, `Compat.String` is an alias to `Base.ByteString` https://github.com/JuliaLang/Compat.jl/blob/v0.26.0/src/Compat.jl#L950 which is a Union: https://github.com/JuliaLang/julia/blob/v0.4.7/base/boot.jl#L259

Since Union or Abstract-ly typed fields in types are bad for performance, the best option would be to leave the 0.4-compatible version of this package alone, change the minimum julia version in REQUIRE to 0.5 and re-tag as a new minor package version, instead of also releasing this change on Julia 0.4.

**Comment by juliohm** [comment]
Will REQUIRE Julia v0.5, delete the tag and create a new tag with a bumped minor version.
